src/app/common/transactions/broadcast-transaction.ts

```diff
@@ -13,7 +14,10 @@ interface BroadcastTransactionOptions {
 export async function broadcastTransaction(options: BroadcastTransactionOptions) {
   const { txRaw, serialized, isSponsored, attachment, networkUrl } = options;
 
-  if (isSponsored) return { txRaw };
+  if (isSponsored) {
+    await delay(1000);
```

Comment:
Should I use your pattern here @kyranjamie and create a new function that wraps this with a more descriptive name?

Reply:
Was going to comment here indeed. It does depend, of course, but if you can describe with a reasonable length method name why this happens, yes definitely.